src/test/java/uk/gov/pay/ledger/transaction/resource/TransactionResourceIT.java

```diff
@@ -187,6 +187,148 @@ public void shouldSearchUsingAllFieldsAndReturnAllFieldsCorrectly() {
                 .body("_links.next_page.href", containsString("v1/transaction?account_id=" + gatewayAccountId + "&from_date=2000-01-01T10%3A15%3A30Z&to_date=2100-01-01T10%3A15%3A30Z&email=example.org&reference=reference&cardholder_name=smith&first_digits_card_number=123456&last_digits_card_number=1234&card_brand=visa%2Cmastercard&state=submitted&transaction_type=PAYMENT&page=3&display_size=2"));
     }
 
+    @Test
+    public void shouldSearchCorrectlyUsingPaymentStates() {
+        String gatewayAccountId = "1";
+        TransactionFixture createdTransaction = aTransactionFixture()
+                .withTransactionType("PAYMENT")
+                .withState(TransactionState.CREATED)
+                .withDefaultCardDetails()
+                .withGatewayAccountId(gatewayAccountId)
+                .withDefaultTransactionDetails()
+                .withCreatedDate(ZonedDateTime.now())
+                .insert(rule.getJdbi());
+
+
+        TransactionFixture submittedTransaction = aTransactionFixture()
+                .withTransactionType("PAYMENT")
+                .withState(TransactionState.SUBMITTED)
+                .withDefaultCardDetails()
+                .withGatewayAccountId(gatewayAccountId)
+                .withDefaultTransactionDetails()
+                .withCreatedDate(ZonedDateTime.now().minusHours(1))
+                .insert(rule.getJdbi());
+
+        TransactionFixture successTransaction = aTransactionFixture()
+                .withTransactionType("PAYMENT")
+                .withState(TransactionState.SUCCESS)
+                .withDefaultCardDetails()
+                .withGatewayAccountId(gatewayAccountId)
+                .withDefaultTransactionDetails()
+                .insert(rule.getJdbi());
+
+        transactionFixture.insert(rule.getJdbi());
+        given().port(port)
+                .contentType(JSON)
+                .get("/v1/transaction?" +
+                        "account_id=" + gatewayAccountId +
+                        "&page=1" +
+                        "&display_size=5" +
+                        "&payment_states=created,submitted"
+                )
+                .then()
+                .statusCode(Response.Status.OK.getStatusCode())
+                .contentType(JSON)
+                .body("results[0].transaction_id", is(createdTransaction.getExternalId()))
+                .body("results[1].transaction_id", is(submittedTransaction.getExternalId()))
+                .body("results[1].transaction_id", is(nullValue()));
```

Comment:
Shouldn't this line be:
```suggestion
                .body("results[2].transaction_id", is(nullValue()));
```
?

That's what's breaking the build.

Reply:
yep